src/Game/zRumble.cpp

```diff
@@ -1,24 +1,156 @@
+#include "../Core/x/xPad.h"
+
 #include "zRumble.h"
+#include "zGlobals.h"
 
 #include <types.h>
 
+extern float32 lbl_803CDE30; // 0.2f
+extern float32 lbl_803CDE34; // 0.1f
+extern float32 lbl_803CDE38; // 0.06f
+extern float32 lbl_803CDE3C; // 5.0f
+extern float32 lbl_803CDE40; // 0.5f
+extern float32 lbl_803CDE44; // 0.3f
+extern float32 lbl_803CDE48; // 0.05f
+extern float32 lbl_803CDE4C; // 0.15f
+extern float32 lbl_803CDE50; // 0.35f
+extern float32 lbl_803CDE54; // 0.6f
+extern float32 lbl_803CDE58; // 0.04f
+extern float32 lbl_803CDE5C; // 0.09f
+extern float32 lbl_803CDE60; // 0.0f
+extern float32 lbl_803CDE64; // 1.0f
+
+#if 1
+
 // func_800ACDE0
 #pragma GLOBAL_ASM("asm/Game/zRumble.s", "zRumbleStart__Fi16_tagSDRumbleTypeP4xEnt")
 
-// func_800AD030
-#pragma GLOBAL_ASM("asm/Game/zRumble.s", "zRumbleStart__Fi16_tagSDRumbleType")
+#else
+
+// Another one that should work, but messes with jump tables. 800acdf0 and 800ace18 are the only ones in the function that don't work.
```

Comment:
It's appreciated that you're deecompiling these switch functions. Although I think your energy would be best spent elsewhere until we figure out how to approach decompiling switch statements with jump tables, but that's up to you.